[jaejayrome] iP

jaejayrome opened the pull request:
# Pan Chatbot

Pan Chatbot is a _revolutionary_ chatbot that allows you to keep track of your tasks!

* perfect for typing individuals 🥇 
  * simple text commands 💯 
  * ~~difficult~~ **easy** syntax
* easily portable ❗ 
* no network required! 👍 

Features Implemented:
- [x] Add, Delete, Mark and Unmark Deadlines
- [x] Add, Delete, Mark and Unmark Events
- [ ] GUI Representation

To get started, you would have to: 
1. clone the repo [here] (https://github.com/jaejayrome/ip).
2. Run Command in Terminal `cd ip`
3. Run Command in Terminal `./gradlew run`

In case you missed out, these are the commands to run:
```bash
cd ip
./gradlew run
```

<blockquote>
So what are you waiting for? Get Started Today With Pan and Improve your Productivity!
</blockquote>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jaejayrome committed:
chore: add *.class to .gitignore

jaejayrome committed:
feat: Level-0:Rename, Greet, Exit

jaejayrome committed:
feat: Level-1 Echo

jaejayrome committed:
feat: Level-2 Add List

jaejayrome committed:
feat: added extra task class

jaejayrome committed:
feat: added getters and setters

jaejayrome committed:
chore: reformatted Pan.java to include Task instances

jaejayrome committed:
feat: added regex pattern checking and mark unmark feature

jaejayrome committed:
chore: fixed line breaks in Task.java

jaejayrome committed:
feat: added 3 child classes and finished ToDos

jaejayrome committed:
feat: finished impl for Deadlines.java

jaejayrome committed:
feat: finished impl for Events.java

jaejayrome committed:
feat: added getters and setters for child classes

jaejayrome committed:
feat: added regex expression for each command to check

jaejayrome committed:
chore: removed unncessary imports

jaejayrome committed:
chore: update .gitignore and update runtest.sh

jaejayrome committed:
chore: fix output styling

jaejayrome committed:
chore: made further changes to the styling to match test case

jaejayrome committed:
feat: added 3 new custom exceptions

jaejayrome committed:
feat: added internal test cases class

jaejayrome committed:
feat: incorporated new additions for exceptions

jaejayrome committed:
chore: added more descriptve error messages

jaejayrome committed:
chore: added break after exception thrown

jaejayrome committed:
chore: removed unnecessary imports

jaejayrome committed:
feat: fixed bug and added delete function

jaejayrome committed:
feat: added 2 enums

jaejayrome committed:
feat: utilised new enums in several files

jaejayrome committed:
Merge branch 'master' of https://github.com/jaejayrome/ip

jaejayrome committed:
feat: implemented save() function

jaejayrome committed:
chore: added .gitignore for pan.txt

jaejayrome committed:
feat: reformatted the child classes attributes

jaejayrome committed:
feat: updated the toString values for conversion to seperate format

jaejayrome committed:
refactor: changed the datetimeformatter of MMM dd yyyy instead

jaejayrome committed:
refactor code into several child classes

jaejayrome committed:
moved all the files within a main package named pan

jaejayrome committed:
deleted pan.java and retested code, added subpackages as well

jaejayrome committed:
deletion of .java files that were in the previous directory

jaejayrome committed:
add the default .class files

jaejayrome committed:
Merge branch 'add-gradle-support'

merge add-gradle-support branch to enable grade usage for iP

jaejayrome committed:
updated the .gitignore to befit the package filepaths

jaejayrome committed:
add gradle support + able to automate and run program using gradle

jaejayrome committed:
add package definitions and gradle configuration

jaejayrome committed:
update .gitignore settings

jaejayrome committed:
add 2 test classes

jaejayrome committed:
remove unused dependencies

jaejayrome committed:
update mark  and unmark for testing

jaejayrome committed:
add specification for test cases and ran with graddle

jaejayrome committed:
add the javadoc commands for all public methods

jaejayrome committed:
add checkstyle to ip

jaejayrome committed:
Merge branch 'master' into branch-A-CodingStandard
merge local main to ensure that javadoc comments are in place

jaejayrome committed:
refactor code to befit checkstyle changes

jaejayrome committed:
add find command for pan

Windofbitter commented on src/main/java/pan/Parser.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Represents the driver method to parse and validate user logic.
+     */
+    public void parseInput() {
```
It is good that even though the  if else is long, but there is no nested if else

Windofbitter commented on src/main/java/pan/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+     *
+     * @param instruction New Task Instance that the user wants to add to the Task List.
+     */
+    public void add(Task instruction) {
```
Maybe we can structure the code by adding spaces

Windofbitter commented on src/main/java/pan/ToDos.java:
```diff
@@ -0,0 +1,29 @@
+ * Pan - Represents the ToDos Class for a ToDos instance
+ * @author Jerome Goh
+ */
+class ToDos extends Task {
```
Maybe we can consider name this class ToDo rather than ToDos

Windofbitter commented on src/main/java/pan/exceptions/InternalTestCases.java:
```diff
@@ -0,0 +1,84 @@
+    public static void testMissingParameters(String instruction) throws MissingParameterException {
+        String brokenCommand = instruction.split(" ")[0].trim();
+        String backCommand = instruction.substring(brokenCommand.length()).trim();
+        switch (brokenCommand) {
```
Good that a default branch is included

Windofbitter reviewed (commented):
Overall, LGTM. I like the comments that are written in a way that is explained what and how neatly, and the coding quality is high.

jaejayrome committed:
updated the correct import statements for test cases

jaejayrome committed:
add javafx dependency and javafx boilerplate to main class

jaejayrome committed:
add new PanLauncher class

jaejayrome committed:
add javafx depenedency to build.gradle

jaejayrome committed:
refactor code to beit javafx syntax

jaejayrome committed:
remove unncessary import and allow Pan to include the refactored Parser

jaejayrome committed:
refactor logic to return to mainwindow and use pan as singleton

simkianseng commented on src/main/java/pan/Storage.java:
```diff
@@ -0,0 +1,33 @@
+            // attempts to create the new file
+            outputFile.createNewFile();
+            StringBuilder sb = new StringBuilder();
+            tasks.stream().forEach(task -> sb.append(task.toString() + "\n"));
```
I like your use of streams to add tasks, it makes the the whole method short. Great job! But perhaps can insert more line breaks to make the code more readable?

simkianseng commented on src/main/java/pan/Pan.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @param instruction The input instruction from the user.
+     */
+    public String parseInput(String instruction) {
```
Perhaps can use a switch case statement here as it seems that there are a lot of cases?

simkianseng commented on src/main/java/pan/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+    public String mark(int index) throws TaskIndexException {
+        if (index > tasks.size() || index < 0) {
+            throw new TaskIndexException("You have entered an invalid index!");
+        } else {
```
Perhaps the else block can be omitted? Since it won't be executed when an exception is thrown. I noticed this in other places as well.

simkianseng reviewed (commented):
Excellent codebase overall! Just some minor issues to improve readability.

jaejayrome committed:
add assertions at important methods

jaejayrome committed:
Merge pull request #1 from jaejayrome/branch-A-Assertions

add assertions at important methods

jaejayrome committed:
MainWindow class: add Javadoc comments for clarity and maintainability

MainWindow.java file contains methods for initialising the scrollPane and dialogContainer, handling user input, and setting the Pan object.

To improve code readability and maintainability, it's necessary to add Javadoc comments to describe the purpose and functionality of the methods.

Add Javadoc comments to the initialize(), setPan(), and handleUserInput() methods.

Javadoc comments provide a standardized way to document code, making it easier for developers to understand how methods work, what parameters they accept, and what they return.

No other changes were made to the code beyond adding Javadoc comments.

jaejayrome committed:
refactor code by adding more javadoc comments, and removing trailing white spaces and overly long lines

jaejayrome committed:
Merge pull request #2 from jaejayrome/A-CodeQuality

A code quality

jaejayrome committed:
TaskList class: add sort function and command to TaskList class

TaskList class lacks a sort command that enable users to sort the tasks based on lexicographical order.

To improve usability, it's necessary to add a sort command to the Pan.

Implemented a sort function within the TaskList class that allows sorting tasks based on lexicographical order. Additionally, refactored the List method to ensure that DRY principle is being adhered to, this would also affect the Pan class.

By incorporating a sort command directly into the TaskList class, users can efficiently organise their tasks without needing external tools or libraries. This approach enhances the overall usability and maintainability of the codebase.

Have made other miscellaneous changes by adding Javadoc comments for the new sort function being added.

jaejayrome committed:
Merge pull request #3 from jaejayrome/branch-C-Sort

TaskList class: add sort function and command to TaskList class

jaejayrome committed:
add persistent storage field for pan

jaejayrome committed:
refine test cases to suit new requirements

jaejayrome committed:
fix checkstyle and javadoc errors

jaejayrome committed:
add ip showcase screenshot

jaejayrome committed:
update readme for the product page
